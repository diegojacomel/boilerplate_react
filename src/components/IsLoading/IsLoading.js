// Modules
import React, { Fragment } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const IsLoadingWrapper = styled('div')`
    display: flex;
    justify-content: ${props => props.centerIcon ? 'center' : null};
    height: 70px;
`

const StyledSpinner = styled.svg`
    position: absolute;
    animation: rotate 2s linear infinite;
    width: 50px;
    height: 50px;
    margin: ${props => props.theme.spacing.sm};
    
    & .path {
        stroke: ${({ theme, color }) => theme.colors[color]};
        stroke-linecap: round;
        animation: dash 1.5s ease-in-out infinite;
    }
    
    @keyframes rotate {
        100% {
            transform: rotate(360deg);
        }
    }
    @keyframes dash {
        0% {
            stroke-dasharray: 1, 150;
            stroke-dashoffset: 0;
        }
        50% {
            stroke-dasharray: 90, 150;
            stroke-dashoffset: -35;
        }
        100% {
            stroke-dasharray: 90, 150;
            stroke-dashoffset: -124;
        }
    }
`;

const IsLoading = ({ isLoading, centerIcon, color }) => (
    <Fragment>
        {isLoading
            ?
            <IsLoadingWrapper centerIcon={centerIcon} id="testIsLoading">
                <StyledSpinner viewBox="0 0 50 50" isLoading={isLoading} color={color} id="testIsLoadingSpinner">
                    <circle
                        className="path"
                        cx="25"
                        cy="25"
                        r="20"
                        fill="none"
                        strokeWidth="5"
                    />
                </StyledSpinner>
            </IsLoadingWrapper>
            :
            null
        }
    </Fragment>
)

IsLoading.propTypes = {
    isLoading: PropTypes.bool,
    centerIcon: PropTypes.bool,
    color: PropTypes.string
}

IsLoading.defaultProps = {
    isLoading: false,
    centerIcon: true,
    color: 'primary'
}

export default IsLoading;