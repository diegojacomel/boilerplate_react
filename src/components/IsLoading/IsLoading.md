### Basic usage

```js
<IsLoading
    isLoading={true}
    centerIcon={false}
    color="primary"
/>
```

#### **isLoading**

The isLoading attribute specifies if will be showing the loader.

type: bool

#### **centerIcon**

The centerIcon attribute specifies where the component will be rendered.

type: bool