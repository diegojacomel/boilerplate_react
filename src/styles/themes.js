// Scale
import { pxToRem } from '../utils/scale';

// Themes
const primaryTheme = {
    colors: {
        primary: "#1a81ff",
        danger: "#ff3333",
        info: "#05cdfa",
        success: "#159d79",
        default: "#373f49",
        white: "#ffffff",
        black: "#1a1a1a",
        greyLight: "#f2f2f2",
        greyMedium: "#999999",
        greyDark: "#333333",
        hover: {
            primary: "#54a0ff",
            danger: "#ff6b6b",
            info: "#48dbfb",
            success: "#18bb90",
            default: "#576574",
            white: "#ffffff",
            black: "#000000"
        },
        active: {
            primary: "#0069ed",
            danger: "#9e0000",
            info: "#05b2d8",
            success: "#11775c",
            default: "#111111",
            white: "#f2f2f2",
            black: "#1a1a1a"
        }
    },
    spacing: {
        none: "0",
        xxs: pxToRem(3),
        xs: pxToRem(5),
        xsm: pxToRem(8),
        sm: pxToRem(10),
        md: pxToRem(15),
        lg: pxToRem(20),
        xlg: pxToRem(25),
        xg: pxToRem(30),
        xxg: pxToRem(50)
    },
    fontSize: {
        none: "0",
        xs: pxToRem(12),
        sm: pxToRem(14),
        md: pxToRem(16),
        lg: pxToRem(18),
        xg: pxToRem(20),
    },
    rounded: {
        none: "0",
        sm: "3px",
        mdLeft: "5px 0 0 5px",
        mdRight: "0 5px 5px 0",
        md: "5px",
        lg: "10px",
        full: "100%"
    }
}

export {
    primaryTheme
}