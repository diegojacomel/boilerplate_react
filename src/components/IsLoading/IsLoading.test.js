// Modules
import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

// Theme
import { primaryTheme } from '../../styles/themes'; 

// Component
import IsLoading from './IsLoading';

describe('IsLoading', () => {
    const wrapper = mount(<IsLoading />);

    test('It exist', () => {
        expect(wrapper).toExist();
    })

    test('Have props', () => {
        expect(wrapper).toHaveProp('centerIcon');
        expect(wrapper).toHaveProp('isLoading');
        expect(wrapper).toHaveProp('color');
    })

    it('Check props', () => {
        expect(wrapper.props().isLoading).toEqual(false);
        expect(wrapper.props().centerIcon).toEqual(true);
        expect(wrapper.props().color).toEqual('primary');
    })

    it('Render correctly', () => {
        const tree = renderer.create(
            <ThemeProvider theme={primaryTheme}>
                <IsLoading
                    isLoading={true}
                    centerIcon={false}
                    color="primary"
                />
            </ThemeProvider>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })


})