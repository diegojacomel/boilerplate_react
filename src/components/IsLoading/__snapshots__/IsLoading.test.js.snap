// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IsLoading Render correctly 1`] = `
<div
  className="sc-bdVaJa OIsdN"
  id="testIsLoading"
>
  <svg
    className="sc-bwzfXH cgppRG"
    color="primary"
    id="testIsLoadingSpinner"
    viewBox="0 0 50 50"
  >
    <circle
      className="path"
      cx="25"
      cy="25"
      fill="none"
      r="20"
      strokeWidth="5"
    />
  </svg>
</div>
`;
